tests/testthat/test-partial-eval.R

```diff
@@ -78,10 +92,72 @@ test_that("across() translates NULL", {
   expect_snapshot(lf %>% mutate(across(a:b)))
 })
 
-test_that("can control names", {
-  lf <- lazy_frame(a = 1, b = 2)
-  out <- lf %>% summarise(across(a:b, c("log", "exp"), .names = "{.fn}_{.col}"))
-  expect_equal(colnames(out), c("log_a", "exp_a", "log_b", "exp_b"))
+test_that("across() can rename", {
+  expect_equal(
+    lazy_frame(x = 1:10) %>%
+      mutate(across(.cols = c(y = x), .fns = as.character)) %>%
+      op_vars(),
+    c("x", "y")
+  )
+})
+
+test_that("across() works with .cols = character()", {
+  lf <- lazy_frame(x = 1:10)
+
+  expect_equal(
+    lf %>% mutate(across(.cols = character(), .fns = as.character)),
+    lf
+  )
+})
+
+test_that("partial_eval_dots() names automatically", {
+  expect_named(
+    partial_eval_dots(quos(across(everything())), c("a", "b")),
+    c("a", "b")
+  )
+})
+
+test_that("across() correctly names output columns", {
+  gf <- lazy_frame(x = 1, y = 2, z = 3, s = "") %>% group_by(x)
+
+  expect_equal(
+    summarise(gf, across()) %>% op_vars(),
+    c("x", "y", "z", "s")
+  )
+  expect_equal(
+    summarise(gf, across(.names = "id_{.col}")) %>% op_vars(),
+    c("x", "id_y", "id_z", "id_s")
+  )
+  expect_equal(
+    summarise(gf, across(1:2, mean)) %>% op_vars(),
+    c("x", "y", "z")
+  )
+  expect_equal(
+    summarise(gf, across(1:2, mean, .names = "mean_{.col}")) %>% op_vars(),
+    c("x", "mean_y", "mean_z")
+  )
+  expect_equal(
+    summarise(gf, across(1:2, list(mean = mean, sum = sum))) %>% op_vars(),
+    c("x", "y_mean", "y_sum", "z_mean", "z_sum")
+  )
+  expect_equal(
+    summarise(gf, across(1:2, list(mean = mean, sum = sum), .names = "{.fn}_{.col}")) %>% op_vars(),
+    c("x", "mean_y", "sum_y", "mean_z", "sum_z")
+  )
+
+  skip("not yet decided")
+  expect_equal(
```

Comment:
Here, `dbplyr` is smarter than `dplyr` and automatically creates names. What do you prefer?